include/dxc/DxilContainer/DxcContainerBuilder.h

```diff
@@ -66,6 +69,12 @@ class DxcContainerBuilder : public IDxcContainerBuilder {
   const char *m_warning;
   bool m_RequireValidation;
   bool m_HasPrivateData;
+  // Function to compute hash when valid dxil container is built
+  // This is nullptr if loaded container has invalid hash
+  HASH_FUNCTION_PROTO *m_HashFunction;
```

Comment:
Why does this need to be stored?  Seems that it is only used in HashAndUpdate.  If there are ways of using DxcContainerBuilder that don't result in HashAndUpdate being called then this means we're always re-hashing existing content to determine the hash function even though we don't need it.

Reply:
Ah, so the idea is that this is only set in DxcContainerBuilder::Load.  I assumed when I was looking that this must be the constructor because I can't find anywhere else that sets m_HashFunction - so what sets it to nullptr in the non-load cases?  I see that there's a DxcContainerBuilder::Init() function in the header that initializes the other members.